### DISCUSSION TITLE
Set display gamma

### DISCUSSION THREAD

**Discussion by toysonlocation** [airsdk/Adobe-Runtime-Support#2028] [General]
For iOS/Android (or even macOS/Windows), is there a way to dynamically set the gamma/color output of the app as a whole?

**Comment by 2jfw** [comment]
I would not know any other way than placing one (or many) `Sprite` with a specific `blendMode` and `Color` on the top of the `DisplayList`.
Though you end up slightly losing performance, I think the decrease isn't actually that high..

**Comment by JohnBlackburne** [comment]
To do it properly no – this is under the control of the OS, probably impossible on mobile, probably requiring low level messing with video APIs on desktop.

To fake it you could add a colour transformation to your rendering, based on the gamma effect you want to achieve. The most efficient way would be to add the code for it to fragment shaders, using AGAL code, treating the output colour as input to a formula that implements the gamma affect. You need to do this for every fragment shader your app uses, creating modified versions. You can then add these to your objects with a custom MeshEffect in Starling. You can make them dynamic by either switching them in and out or by incorporating the dynamism into the code, by e.g. passing a parameter to the shader that indicates the amount of gamma desired.
